### DISCUSSION TITLE
v1.104.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#9434] [Announcements]
# v1.104.0

> [!CAUTION] 
> # EXTERNAL LIBRARY EDITABILITY
> For external library users, you can now manage your assets directly from Immich's user interface, i.e. you can edit date/time, location information, and **delete** from the web and the mobile app. 
>
> If you don't want Immich to handle those operations, please make sure to have the read-only, i.e., `:ro` flag on your mount point in the `docker-compose.yml` file.

## Live stream - Immich Q&A ft. Eron Wolf & Louis Rossmann from FUTO

Last week, we had a live stream to answer your questions about Immich joining FUTO. You can watch the VOD [here](https://www.youtube.com/watch?v=cwz2iZwYpgg).

## Highlights
Welcome to release `v1.104.0` of Immich. This release includes new features, bug fixes, and improvements. The core team will now focus on tasks to bring us to a stable release. The highlights of this release can be found below.

- Editability of external library
- Notification foundation - SMTP Email

### Editability of external library

Yes, we've heard you guys! You can now edit date/time and location information and even __delete__ assets in an external library directly from Immich. If you don't want to use this feature, you should make sure to have the `:ro` flag on your docker mount in the `docker-compose.yml` file.

### Notification - SMTP Email

Thanks, @hitech95, for kickstarting this feature,@jrasm91, and @danieldietzler for finishing it. We now have basic notification built-in, using email. The first mechanism that uses email is to send an email to a new user with information about the account and the instance they were invited to.

<img src="https://github.com/immich-app/immich/assets/27055614/949cba85-d3f1-4cd3-b246-a6f5fb5d3ae8" width="600"/>

<img src="https://github.com/immich-app/immich/assets/27055614/1066a574-9c72-4375-af51-58e0284574e8" width="600"/>

### ⚠️ Breaking Changes
* feat(server): read-write external assets by @jrasm91 in https://github.com/immich-app/immich/pull/9235
### 🗄️ Server
* feat(server): email notifications by @hitech95 in https://github.com/immich-app/immich/pull/8447
* feat(server): immich checksum header by @jrasm91 in https://github.com/immich-app/immich/pull/9229
* fix(server, web): reassigning faces by @martabal in https://github.com/immich-app/immich/pull/9265
* fix(server): show partners archived photos on mobile timeline by @alextran1502 in https://github.com/immich-app/immich/pull/9194
* refactor: create album users by @jrasm91 in https://github.com/immich-app/immich/pull/9315
* feat(server) Random Endpoint use querybuilder and return exifInfo by @JW-CH in https://github.com/immich-app/immich/pull/9301
* feat(server): optimize person thumbnail generation by @mertalev in https://github.com/immich-app/immich/pull/7513
* chore(server): remove unused deps by @jrasm91 in https://github.com/immich-app/immich/pull/9334
* refactor(server): auth route metadata by @jrasm91 in https://github.com/immich-app/immich/pull/9344
* perf: cache `getConfig` by @mertalev in https://github.com/immich-app/immich/pull/9377
* perf: cache transcoding devices by @mertalev in https://github.com/immich-app/immich/pull/9381
* chore(server): less rigid transcoding tests by @mertalev in https://github.com/immich-app/immich/pull/9401
* feat(server, web): include pictures of shared albums on map by @andreasgerstmayr in https://github.com/immich-app/immich/pull/7439
* chore(server): email notification button png by @alextran1502 in https://github.com/immich-app/immich/pull/9423
* fix(server): put system config by @jrasm91 in https://github.com/immich-app/immich/pull/9425
### 📱 Mobile
* refactor(mobile): entities and models by @alextran1502 in https://github.com/immich-app/immich/pull/9182
* fix(mobile): show all places don't show all places by @alextran1502 in https://github.com/immich-app/immich/pull/9193
* feat(mobile): play motion video with long press gesture support by @jzhangdev in https://github.com/immich-app/immich/pull/6543
* refactor(mobile): services and providers by @alextran1502 in https://github.com/immich-app/immich/pull/9232
* fix(mobile): Reset dropdown values in search by location on mobile by @Tushar-Harsora in https://github.com/immich-app/immich/pull/9261
* refactor(mobile): pages by @alextran1502 in https://github.com/immich-app/immich/pull/9246
* chore(mobile) Linked pt-BR.json with Mobile app and localizely.yml by @Tushar-Harsora in https://github.com/immich-app/immich/pull/9281
* Localizely: Translations update by @alextran1502 in https://github.com/immich-app/immich/pull/9287
* fix(mobile): allow select empty album as backup album by @alextran1502 in https://github.com/immich-app/immich/pull/9290
* refactor(mobile): widgets by @alextran1502 in https://github.com/immich-app/immich/pull/9291
* fix(mobile): add user to album using new dto property by @alextran1502 in https://github.com/immich-app/immich/pull/9312
* fix(mobile): fix ca-CA.json file name by @aryiu in https://github.com/immich-app/immich/pull/9288
* fix(mobile): Fix asset selector title bar text by @ddshd in https://github.com/immich-app/immich/pull/9273
* chore(deps): update mobile by @renovate in https://github.com/immich-app/immich/pull/7229
* feat(mobile): Add pull to refresh to sharing page by @ddshd in https://github.com/immich-app/immich/pull/9274
* Revert "feat(android) Check server is reachable before starting background backup (#8989)" by @alextran1502 in https://github.com/immich-app/immich/pull/9324
* fix(mobile): timeline slider no longer disappears by @ConnerWithAnE in https://github.com/immich-app/immich/pull/9358
* refactor(mobile): remove shared module by @alextran1502 in https://github.com/immich-app/immich/pull/9363
* Localizely: Translations update by @alextran1502 in https://github.com/immich-app/immich/pull/9432
### 🖥️ Web
* feat(web,a11y): replace IconButton with CircleIconButton by @ben-basten in https://github.com/immich-app/immich/pull/9153
* refactor(web): search people by @martabal in https://github.com/immich-app/immich/pull/9082
* fix(web): Prevent DELETE key from clearing DateInput in modal by @uniqueeest in https://github.com/immich-app/immich/pull/8846
* fix(web): missing add photos button in album when viewed as owner by @alextran1502 in https://github.com/immich-app/immich/pull/9183
* fix(web): Configure web's immich sdk to use event.fetch by @midzelis in https://github.com/immich-app/immich/pull/9233
* feat(web): determine duplication of upload on client by @truongsinh in https://github.com/immich-app/immich/pull/8825
* feat(web): expand usage of CircleIconButton by @ben-basten in https://github.com/immich-app/immich/pull/9256
* fix(web):  Individual Photo Sharing with Album Link #8887 by @iamamansharma in https://github.com/immich-app/immich/pull/9264
* fix(web): click on face in detail-panel by @martabal in https://github.com/immich-app/immich/pull/9271
* fix(web): stream 360 video instead of fetching entire video by @TapuCosmo in https://github.com/immich-app/immich/pull/9279
* fix(web): allow deselecting all assets from select bar by @Snowknight26 in https://github.com/immich-app/immich/pull/9320
* fix(web): Selected face in search filter doesn't show border highlight when hovered in light theme by @Tushar-Harsora in https://github.com/immich-app/immich/pull/9348
* fix(web): fix logo size on shared pages by @Snowknight26 in https://github.com/immich-app/immich/pull/9346
* fix(web): multiple fixes for people by @martabal in https://github.com/immich-app/immich/pull/9343
* fix(web): autofocus change name field by @alextran1502 in https://github.com/immich-app/immich/pull/9376
* fix(web): Fixed video unmutes when scrubbing by @Tushar-Harsora in https://github.com/immich-app/immich/pull/9382
* fix(web): page down shortcut by @martabal in https://github.com/immich-app/immich/pull/9397
* chore(web): icon consistency and improvements by @NicholasFlamy in https://github.com/immich-app/immich/pull/9403
* feat(web): hide all unnamed by @martabal in https://github.com/immich-app/immich/pull/9378
### 📓 Documentation
* chore: upgrade docusaurus-preset-openapi to remove axios by @benmccann in https://github.com/immich-app/immich/pull/9167
* Docs: New Futo milestones by @aviv926 in https://github.com/immich-app/immich/pull/9211
* chore(readme): add Brazilian Portuguese readme by @ryanrpj in https://github.com/immich-app/immich/pull/7107
* chore: Remove donation info from readme_i18n by @bo0tzz in https://github.com/immich-app/immich/pull/9225
* chore: Remove sponsor link from homepage by @bo0tzz in https://github.com/immich-app/immich/pull/9226
* chore(docs): Link to helm chart README by @bo0tzz in https://github.com/immich-app/immich/pull/9238
* docs: update post-install-steps documentation by @g-bulgarit in https://github.com/immich-app/immich/pull/9239
* Specify PowerShell for win backup/restore by @mmomjian in https://github.com/immich-app/immich/pull/9262
* docs: add custom map styles community guide by @vovchykbratyk in https://github.com/immich-app/immich/pull/9311
* chore: arabic translations by @Me-Mr in https://github.com/immich-app/immich/pull/9325
* docs: update portainer.md by @mtwb47 in https://github.com/immich-app/immich/pull/9337
* Chore: Add Arabic translations to readme by @Me-Mr in https://github.com/immich-app/immich/pull/9342
* docs: backup restore - add warning to not backup DB_DATA_LOCATION by @mmomjian in https://github.com/immich-app/immich/pull/9353
* docs: adds hyperlinks for other readme translations by @Me-Mr in https://github.com/immich-app/immich/pull/9345
* chore: put back warning banner by @alextran1502 in https://github.com/immich-app/immich/pull/9362
* docs: warn against use on NTFS / WSL by @mmomjian in https://github.com/immich-app/immich/pull/9371
* docs: add discord link by @danieldietzler in https://github.com/immich-app/immich/pull/9379
* docs: small changes&updates by @aviv926 in https://github.com/immich-app/immich/pull/9059
* Update README_nl_NL.md with missing parts included on the english readme by @Deedikjupijn in https://github.com/immich-app/immich/pull/9399
* docs: add info about database checksums by @mmomjian in https://github.com/immich-app/immich/pull/9391
* Update portainer install docs by @mmomjian in https://github.com/immich-app/immich/pull/9421
* Update external-library.md by @FedericoCalzoni in https://github.com/immich-app/immich/pull/9420
### 🔨 Maintenance
* chore(deps): update dependency supertest to v7 by @renovate in https://github.com/immich-app/immich/pull/9156
* chore(deps): update node.js to 142644a by @renovate in https://github.com/immich-app/immich/pull/9165
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/9168
* chore(deps): update node.js to 7a91aa3 by @renovate in https://github.com/immich-app/immich/pull/9166
* fix(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/9169
* chore(deps): update mambaorg/micromamba:bookworm-slim docker digest to f4c5c3c by @renovate in https://github.com/immich-app/immich/pull/9181
* chore(deps): update dependency ruff to v0.4.2 by @renovate in https://github.com/immich-app/immich/pull/9186
* chore(deps): update base-image to v20240430 (major) by @renovate in https://github.com/immich-app/immich/pull/9179
* chore: fvm and node dependency updates and pinning by @zackpollard in https://github.com/immich-app/immich/pull/9203
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/9292
* fix(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/9293
* chore(deps): update base-image to v20240507 (major) by @renovate in https://github.com/immich-app/immich/pull/9303
* fix(deps): update dependency exiftool-vendored to v26 by @renovate in https://github.com/immich-app/immich/pull/9294
* chore(deps): update dependency @types/nodemailer to v6.4.15 by @renovate in https://github.com/immich-app/immich/pull/9309
* chore: remove funding file by @alextran1502 in https://github.com/immich-app/immich/pull/9319
* chore(deps): update node.js to fe31b16 by @renovate in https://github.com/immich-app/immich/pull/9318
* chore(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/9314
* chore(deps): update node.js to fac6f74 by @renovate in https://github.com/immich-app/immich/pull/9317
* chore(deps): bump tj-actions/verify-changed-files from 19 to 20 by @dependabot in https://github.com/immich-app/immich/pull/9280
* chore: move test assets submodule by @jrasm91 in https://github.com/immich-app/immich/pull/9321
* chore(deps): update prom/prometheus docker digest to 5c43564 by @renovate in https://github.com/immich-app/immich/pull/9341
* feat(server): enable data page checksums by @mmomjian in https://github.com/immich-app/immich/pull/9384
* fix(deps): update dependency nestjs-otel to v6 by @renovate in https://github.com/immich-app/immich/pull/9415

## New Contributors
* @uniqueeest made their first contribution in https://github.com/immich-app/immich/pull/8846
* @hitech95 made their first contribution in https://github.com/immich-app/immich/pull/8447
* @jzhangdev made their first contribution in https://github.com/immich-app/immich/pull/6543
* @ryanrpj made their first contribution in https://github.com/immich-app/immich/pull/7107
* @g-bulgarit made their first contribution in https://github.com/immich-app/immich/pull/9239
* @Tushar-Harsora made their first contribution in https://github.com/immich-app/immich/pull/9261
* @TapuCosmo made their first contribution in https://github.com/immich-app/immich/pull/9279
* @vovchykbratyk made their first contribution in https://github.com/immich-app/immich/pull/9311
* @aryiu made their first contribution in https://github.com/immich-app/immich/pull/9288
* @Me-Mr made their first contribution in https://github.com/immich-app/immich/pull/9325
* @mtwb47 made their first contribution in https://github.com/immich-app/immich/pull/9337
* @Deedikjupijn made their first contribution in https://github.com/immich-app/immich/pull/9399
* @andreasgerstmayr made their first contribution in https://github.com/immich-app/immich/pull/7439
* @FedericoCalzoni made their first contribution in https://github.com/immich-app/immich/pull/9420

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.103.1...v1.104.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.104.0'>v1.104.0</a>.</em>

**Comment by fwsmit** [comment]
It seems the link in the email would only open in the browser, not in the app. It would be nice if there was a button in the email to open the app directly with the right server filled in. This would probably require to use an immich.app domain with some hint in it.
This is one of the ways in which self hosting gives a worse user experience. It's hard or sometimes impossible to get it right though

**Reply by jrasm91** [reply]
I think we are going to look into this

**Comment by alex-007** [comment]
Does someone know how to set up SMTP server, so gmail will agree to get emails from it?

**Reply by Me-Mr** [reply]
Gmail might be the easiest to setup https://support.google.com/a/answer/176600?hl=en

**Reply by alex-007** [reply]
I don't have a `Google Workspace`. Also I don't want to set my own email as credentials (it will store the sent emails to my Sent folder in Gmail, which will brake the privacy of my relatives)

**Reply by alextran1502** [reply]
You don't need Google Workspace, you can use App Password instead https://mailtrap.io/blog/gmail-smtp/

**Comment by nicomerl** [comment]
I changed some of the Meta-Data of pictures and videos in the past. If i am right i read somewhere that, right now, the Data is not changed but stored somewhere else. 

Is this still the case? Would it be possible to change the original files metadata? 

I would like to have all my pictures with the correct metadata in den /upload -Folder if i need to restore.

**Reply by haldi4803** [reply]
THIS!
When i read "**EXTERNAL LIBRARY EDITABILITY**" i tought it would actually be possible to edit exif Metadata of the Pictures!
But actually NO. it doesn't.
It simply adds an additional .xmp file with the same name that has the changes inside.

```
<?xpacket begin='﻿' id='W5M0MpCehiHzreSzNTczkc9d'?>
<x:xmpmeta xmlns:x='adobe:ns:meta/' x:xmptk='Image::ExifTool 12.84'>
<rdf:RDF xmlns:rdf='http://www.w3.org/1999/02/22-rdf-syntax-ns#'>

 <rdf:Description rdf:about=''
  xmlns:pdf='http://ns.adobe.com/pdf/1.3/'>
  <pdf:CreationDate>2024-04-04T13:10:00.000+09:00</pdf:CreationDate>
 </rdf:Description>
</rdf:RDF>
</x:xmpmeta>
<?xpacket end='w'?>
```

So if i send the file to a friend after editing Timezone or Date he will still have the file with the wrong infos.

But after looking into it, it seems like metadata editing is a PITA, so i can see why they opted for this way.

**Reply by mertalev** [reply]
I don't think this is any different than for uploaded assets. We have a general rule of not messing with the original file.

**Comment by MickLesk** [comment]
@alextran1502 

i have now changed some german translations :-) that sounds a bit better in word and writing. (localizly)
(for the next version :P)

**Comment by weexceptions** [comment]
Browser upload seems broken lables
It shows UPLOAD.... instead of speed 
&
Success message: Uploaded(Uploading...l

**Comment by Solarunit** [comment]
I wish I could separate all items in External library from the Photos. For example by showing External library items in new library category:

- Favorites
- Albums
- **External library items** :arrow_left: 
- Trash
- ...
